lbrynet/database/storage.py

```diff
@@ -0,0 +1,457 @@
+    reannounce = max(min_reannounce_time,
+                     queue_size * single_announce_duration)
+    return time.time() + reannounce
+
+
+def rerun_if_locked(f):
+    def rerun(err, *args, **kwargs):
+        log.error("Failed to execute (%s): %s", err, args)
+        if err.check(sqlite3.OperationalError) and err.value.message == "database is locked":
+            log.warning("database was locked. rerunning %s with args %s, kwargs %s",
+                        str(f), str(args), str(kwargs))
+            return task.deferLater(reactor, 0, wrapper, *args, **kwargs)
+        return err
+
+    def wrapper(*args, **kwargs):
+        d = f(*args, **kwargs)
+        d.addErrback(rerun, *args, **kwargs)
+        return d
+
+    return wrapper
+
+
+class SqliteConnection(adbapi.ConnectionPool):
+    def __init__(self, db_path):
+        adbapi.ConnectionPool.__init__(self, 'sqlite3', db_path, check_same_thread=False)
+
+    @rerun_if_locked
+    def runInteraction(self, interaction, *args, **kw):
+        return adbapi.ConnectionPool.runInteraction(self, interaction, *args, **kw)
+
+
+class SQLiteStorage(object):
+    def __init__(self, db_dir):
+        self.db_dir = db_dir
+        self._db_path = os.path.join(db_dir, "lbrynet.sqlite")
+        log.info("connecting to %s", self._db_path)
+        self.db = SqliteConnection(self._db_path)
+
+    def setup(self):
+        def _create_tables(transaction):
+
+            # create table if not exists support (
+            #     support_outpoint text not null primary key,
+            #     claim_id text not null,
+            #     amount real not null,
+            #     valid_at_height real not null
+            # );
+
+            transaction.executescript("""
+            pragma foreign_keys=on;
+            pragma journal_mode=WAL;
+    
+            create table if not exists blob (
+                blob_hash char(96) primary key not null,
+                blob_length integer not null,
+                next_announce_time integer not null,
+                should_announce integer not null default 0,
+                status text not null
+            );
+            
+            create table if not exists stream (
+                stream_hash char(96) not null primary key,
+                sd_hash char(96) not null,
+                stream_key text not null,
+                stream_name text not null,
+                suggested_filename text not null,
+                foreign key(sd_hash) references blob(blob_hash)
+            );
+            
+            create table if not exists stream_blob (
+                stream_hash char(96) not null,
+                blob_hash char(96),
+                position integer not null,
+                iv char(32) not null,
+                primary key (stream_hash, blob_hash),
+                foreign key(stream_hash) references stream(stream_hash),
+                foreign key (blob_hash) references blob(blob_hash)
+            );
```

Comment:
I'll add a commit updating the db migrator at the end of the pr. Since moving the create tables statement out to a class attribute of `SQLiteStorage`, the migrator can use it instead of having a  duplicate create tables query (which is prone to being out of sync)